### PR TITLE
Adds babies.

### PR DESCRIPTION
**Pull request by Schnayy**
I have the power.

### CODE REVIEW

**Comment by Cerebulon** [comment]
die

**Comment by Schnayy** [comment]
> 
> 
> die

You cannot kill me in a way that matters.

File: code/game/objects/items/weapons/baby.dm [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+/obj/item/clothing/suit/baby/verb/baby_emote()
+	set name = "Baby Emote"
+	set category = "Object"
+	set desc = "Click to have your baby perform an emote."
+	set src in usr
+
+	var/message = sanitize(copytext(input(usr, "[name]...", "Baby Emote", null)  as text,1,MAX_MESSAGE_LEN))
+	if(!message)
+		return
+	if ((src.loc == usr && usr.stat == 0))
```
Comment (Atermonera) [inline code/game/objects/items/weapons/baby.dm:46]: `src.loc == usr` is already enforced by the verb only being available if the object is in usr's contents (That is, their inventory, or possibly frankensteined onto them as a limb)

File: code/game/objects/items/weapons/baby.dm [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		for(var/mob/O in (viewers(usr)))
+			O.show_message("<B>[src]</B> [message]")
+
+/obj/item/clothing/suit/baby/black
+	icon_state = "baby-black"
```
Comment (Atermonera) [inline code/game/objects/items/weapons/baby.dm:51]: You could set up additive/multiplicative coloring to get non-human babies from a single base type.

File: code/game/objects/items/weapons/baby.dm [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+/obj/item/clothing/suit/baby/tanned
+	icon_state = "baby-tanned"
```
Comment (Atermonera) [inline code/game/objects/items/weapons/baby.dm:54]: baby_unathi and baby_tajara are defined in baby.dmi, but not in back sprites or referenced here in code.

**Review by Atermonera — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/baby.dm (hunk 2 above)
Comment (Schnayy) [reply code/game/objects/items/weapons/baby.dm:51]: Out of scope for this PR.

File: code/game/objects/items/weapons/baby.dm (hunk 3 above)
Comment (Schnayy) [reply code/game/objects/items/weapons/baby.dm:54]: Unathi babies will not be implemented. Baby tajaran require worn sprites that were not available in this port & a sign-off from tajaran lore maintainer. Would you like them provided now, or later?

File: code/game/objects/items/weapons/baby.dm [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+	w_class = ITEMSIZE_LARGE
+	slot_flags = SLOT_OCLOTHING | SLOT_BACK
+
+/obj/item/clothing/suit/baby/attack_self(mob/living/user as mob)
+	if (user.client)
+		if(user.client.prefs.muted & MUTE_IC)
+			src << "\red You cannot speak in IC (muted)."
+			return
+
+	var/message = sanitize(copytext(input(user, "[name]...", "Baby Emote", null)  as text,1,MAX_MESSAGE_LEN))
```
[the remark this answers is not in the document]
Comment (Schnayy) [reply code/game/objects/items/weapons/baby.dm:15]: Apparently make it work at all, as removing it made emotes fail to display.

**Comment by elgeonmb** [comment]
i step out for five months and this i what happens

**Comment by GeneriedJenelle** [comment]
oh no gurl what u doin. turn back while you can!

**Comment by TeleTubby2** [comment]
make it so you can make baby burgers if you put them in a gibber, i can make sprites if needed

**Comment by Cerebulon** [comment]
> 
> 
> make it so you can make baby burgers if you put them in a gibber, i can make sprites if needed

baby back ribs pls

**Comment by Schnayy** [comment]
> 
> 
> make it so you can make baby burgers if you put them in a gibber, i can make sprites if needed

What the fuck did you just fucking say to do to my baby PR, you little baby? I’ll have you know I graduated top of my class in the Home Economics Department, and I’ve been involved in numerous nanny jobs, and I have over 300 confirmed children. I am trained in childhood education and I’m the top nanny in the entire US Child Services. You are nothing to me but just another target. I will wipe you the fuck out with precise babysitting the likes of which has never been seen before on this Earth, mark my fucking words. You think you can get away with saying that shit to me over the Internet? Think again, child. As we speak I am contacting my secret network of spies across the USA and your IP is being traced right now so you better prepare for the storm, you absolute toddler. The storm that wipes out the pathetic little thing you call your life. You’re fucking dead, kid. I can be anywhere, anytime, and I can swaddle you in over seven hundred ways, and that’s just with my bare hands. Not only am I extensively trained in play time, but I have access to the entire arsenal of the United States Child Protective Services and I will use it to its full extent to wipe your miserable ass off the face of the continent, you little baby. If only you could have known what unholy retribution your little “clever” comment was about to bring down upon you, maybe you would have held your fucking tongue. But you couldn’t, you didn’t, and now you’re paying the price, you goddamn idiot. I will shit fury all over you and you will drown in it. You’re fucking dead, kiddo.

**Comment by Novacat** [comment]
...Would it be at all possible to move the sprites in a way where it can be removed? The main issue is that the worn/held in hand iconfiles are omnibus general purpose thus it would be impossible to remove them without causing future upstream conflicts.

**Comment by Natje0** [comment]
What the fuck

**Comment by nerdass** [comment]
I notice these are coded in as a weapon.

How effective are they in combat?

**Comment by nerdass** [comment]
aE0TOqApOVKQgZhWuUILg1ytNelx4u6ZTJzUmJDbXG0


Feature request, baby armor.

**Comment by Schnayy** [comment]
image

**Pull request closed without merging**